src/lock.ts

```diff
@@ -40,14 +50,23 @@ export class RWLock {
       await Promise.all(this._readP);
       const {promise, resolve} = resolver();
       this._writeP = promise;
-      this._readP = [];
-      return resolve;
+      this._readP.clear();
```

Comment:
I don't think a Set is needed here. You can also do `length = 0` on arrays.

Reply:
A Set will make `this._readP.delete(promise);` more efficient.